lib/LaTeXML/resources/XSLT/LaTeXML-picture-xhtml.xsl

```diff
@@ -139,10 +139,23 @@
         </xsl:choose>
       </xsl:for-each>
       <xsl:choose>
-        <!-- If foreignObject in a DIFFERENT namespace, copy as foreign markup -->
-        <xsl:when test="local-name()='foreignObject'
-                        and not(namespace-uri(child::*) = $SVG_NAMESPACE)">
-          <xsl:apply-templates mode='copy-foreign'/>
+        <xsl:when test="local-name()='foreignObject'">
+          <!-- center the content within foreignObject (note that display:flex does not work on foreignObject so we use two wrappers) -->
+          <xsl:element name="div" namespace="{$html_ns}">
+            <xsl:attribute name="class">ltx_foreignobject_container</xsl:attribute>
+            <xsl:element name="div" namespace="{$html_ns}">
+              <xsl:attribute name="class">ltx_foreignobject_content</xsl:attribute>
+              <!-- If foreignObject in a DIFFERENT namespace, copy as foreign markup -->
+              <xsl:choose>
+                <xsl:when test="not(namespace-uri(child::*) = $SVG_NAMESPACE)">
+                  <xsl:apply-templates mode='copy-foreign'/>
+                </xsl:when>
+                <xsl:otherwise>
+                  <xsl:apply-templates/>
+                </xsl:otherwise>
+              </xsl:choose>
+            </xsl:element>
+          </xsl:element>
```

Comment:
This obscures the previous branch of `foreignObject` (from *non* svg namespace), with the side effect of also wrapping those cases with a pair of `div`s. The new code should have been its own branch below, before the `otherwise`.

Reply:
In hindsight, I actually don't understand the test `not(namespace-uri(child::*) = $SVG_NAMESPACE)`. This is checking whether the first child is not in SVG namespace. I think that maybe you want to check whethere any child is not SVG and if there are non-trivial text nodes.

Reply:
Right, that xpath test tries to say "not any", but ends up saying "not first".

Probably should be closer to:

```xpath
count(child::*[namespace-uri(self) = $SVG_NAMESPACE]) = 0
```

Reply:
And text nodes. Maybe latexml never produces text there, but in principle foreignObject can contain character data.